graphs/views.py

```diff
@@ -300,44 +300,64 @@ def _graphs_page(request, view_type):
+        title_of_graph = None
 
-            title_of_graph = None
+        if 'title' in request.POST:
+            title_of_graph = request.POST['title']
 
-            if 'title' in request.POST:
-                title_of_graph = request.POST['title']
+        if str(request.FILES['graphname'])[-4:] == 'json':
+            upload = 'json'
+        elif str(request.FILES['graphname'])[-4:] == 'gpml':
+            upload = 'gpml'
+        else:
+            upload = 'cyjs'
+
+        if request.POST['email'] == 'Public User':
+            # assign random id generator
+            if upload == 'json':
+                result = db.uploadJSONFile(None, request.FILES['graphname'].read(), title_of_graph)
+            elif upload == 'gpml':
+                result = db.uploadGPMLFile(None, request.FILES['graphname'].read(), title_of_graph)
+                request.POST['layout_name'] = 'gpml'
+                request.POST['loggedIn'] = result['public_user_id']
+                request.POST['public'] = 0
+                request.POST['unlisted'] = 0
+            else:
+                result = db.uploadCyjsFile(None, request.FILES['graphname'].read(), title_of_graph)
 
-            if str(request.FILES['graphname'])[-4:] != "json":
-                upload_json = None
-            
-            if request.POST['email'] == 'Public User':
-                # assign random id generator
-                if upload_json:
-                    result = db.uploadJSONFile(None, request.FILES['graphname'].read(), title_of_graph)
-                else:
-                    result = db.uploadCyjsFile(None, request.FILES['graphname'].read(), title_of_graph)
+            if 'Error' not in result:
+                if upload == 'gpml':
+                    request.POST['points'] = result['default']
+                    temp = save_layout(request, result['public_user_id'], result['title'])
+                context = {'login_form': login_form, 'register_form': register_form, 'Success': result['Success']}
+            else:
+                context = {'login_form': login_form,  'register_form': register_form, 'Error': result['Error']}
+            return render(request, 'graphs/upload_graph.html', context)
+        else:
 
-                if 'Error' not in result:
-                    context = {'login_form': login_form, 'register_form': register_form, 'Success': result['Success']}
-                else:
-                    context = {'login_form': login_form,  'register_form': register_form, 'Error': result['Error']}
-                return render(request, 'graphs/upload_graph.html', context)
+            if upload == 'json':
+                result = db.uploadJSONFile(request.POST['email'], request.FILES['graphname'].read(), title_of_graph)
+            elif upload == 'gpml':
+                result = db.uploadGPMLFile(request.POST['email'], request.FILES['graphname'].read(), title_of_graph)
+                request.POST['layout_name'] = 'gpml'
```

Comment:
Why is layout name hardcoded ? Also why are other parameter initialised the way they are ?